src/erlfmt.erl

```diff
@@ -69,6 +74,30 @@ format_file(FileName, Out) ->
         {error, Error} -> {error, Error}
     end.
 
+-spec format_range(
+    file:name_all(),
+    erlfmt_scan:location(),
+    erlfmt_scan:location()
+) ->
+    {ok, string(), [error_info()]} |
+    {error, error_info()} |
+    {options, [{erlfmt_scan:location(), erlfmt_scan:location()}]}.
+format_range(FileName, StartLocation, EndLocation) ->
+   try
+        {ok, Forms, Warnings} = file_read_forms(FileName),
+        case verify_ranges(Forms, StartLocation, EndLocation) of
+            ok ->
+                InRange = forms_in_range(Forms, StartLocation, EndLocation),
```

Comment:
It looks like `verify_ranges` ends up calling `forms_in_range` as well. Would it make sense to perform this computation only once?

Reply:
Yes, I can move that in so that verify_ranges also returns the range.

Reply:
Pushed a new version where this is not done twice.